src/v/archival/ntp_archiver_service.cc

```diff
@@ -149,12 +173,101 @@ ss::future<> ntp_archiver::upload_loop() {
     }
 }
 
+ss::future<> ntp_archiver::sync_manifest_loop() {
+    ss::lowres_clock::duration backoff = _upload_loop_initial_backoff;
+
+    while (sync_manifest_loop_can_continue()) {
+        auto result = co_await sync_manifest();
+
+        if (result.num_failed != 0) {
+            // The logic in class `remote` already does retries: if we get here,
+            // it means the upload failed after several retries, indicating
+            // something non-transient may be wrong.  Hence error severity.
+            vlog(
+              _rtclog.error,
+              "Failed to upload {} segments out of {}",
+              result.num_failed,
+              result.num_succeded + result.num_failed);
+        } else if (result.num_succeded != 0) {
+            vlog(
+              _rtclog.debug,
+              "Successfuly uploaded {} segments",
+              result.num_succeded);
+        }
+
+        if (!upload_loop_can_continue()) {
+            break;
+        }
+
+        if (result.num_succeded == 0) {
+            // The backoff algorithm here is used to prevent high CPU
+            // utilization when redpanda is not receiving any data and there
+            // is nothing to update. Also, we want to limit amount of
+            // logging if nothing is uploaded because of bad configuration
+            // or some other problem.
+            //
+            // We want to limit max backoff duration
+            // to some reasonable value (e.g. 5s) because otherwise it can
+            // grow very large disabling the archival storage
+            vlog(
+              _rtclog.trace, "Nothing to upload, applying backoff algorithm");
+            co_await ss::sleep_abortable(
```

Comment:
All this backoff logic can be removed for good. It's needed in upload loop because upload loop doesn't have to sleep when it has some data to upload. But if it doesn't have the data the upload loop will busy-wait for data and sleep with backoff is there to prevent that. But here you have a sleep in `sync_manifest` by design so you don't have to use backoff.